generate-manifest

```diff
@@ -0,0 +1,401 @@
+            channel_spec[key] = value[0]
+
+    return channel_spec
+
+
+def parse_content_rating_spec(content_rating_spec_str):
+    content_rating = {}
+
+    attrs = urllib.parse.parse_qs(content_rating_spec_str)
+    for attr_name, attr_value in attrs.items():
+        content_rating[attr_name] = attr_value[0]
+
+    return content_rating
+
+
+def merge_args(catalog_object, cli_args):
+    args = {}
+
+    for key, cli_value in cli_args.items():
+        if key == "channel":
+            args["channels"] = list(map(parse_cli_channel_spec, cli_value))
+            args["channels"].extend(catalog_object.get("channels", []))
+        elif key == "metainfo_content_rating":
+            if cli_value is not None:
+                args[key] = parse_content_rating_spec(cli_value)
+            else:
+                args[key] = catalog_object.get(key, {})
+        else:
+            args[key] = catalog_object.get(key)
+            if cli_value is not None:
+                args[key] = cli_value
+
+    return args
+
+
+def validate_args(args):
+    if len(args["channels"]) == 0:
+        raise ValueError("You must specify at least one channel")
+
+    for channel in args["channels"]:
+        if not channel.get("channel_id"):
+            raise ValueError("Missing channel_id in {}".format(channel))
+
+        if channel.get("origin") == "disk" and not channel.get("directory"):
+            raise ValueError(
+                "origin in 'disk' mode but 'directory' is missing for channel {}".format(
+                    channel
+                )
+            )
+
+
+def main():
+    ap = argparse.ArgumentParser(
+        allow_abbrev=False,
+        description="Generate org.learningequality.Kolibri.Content "
+        "Flatpak manifest for selected content",
+    )
+    ap.add_argument("--verbose", "-v", action="store_true")
+    ap.add_argument(
+        "--id",
+        required=True,
+        help="the ID of the package, it is used as the extension Flatpak ID last segment. "
+        "If the ID is available in extensions-catalog.json, the arguments in that file are "
+        "used and you can still override anything using command line options",
+    )
+    ap.add_argument(
+        "--branch",
+        default="1.0",
+        help="the branch specified in the extension point definition",
+    )
+    ap.add_argument(
+        "--install-kolibri-flatpak",
+        default=False,
+        action="store_true",
+        help="automatically install org.learningequality.Kolibri from Flathub",
+    )
+    ap.add_argument(
+        "--skip-channel-import",
+        default=False,
+        action="store_true",
+        help="skip cleaning up the local Kolibri home directory and channel importing",
+    )
+    ap.add_argument(
+        "--channel",
+        action="append",
+        default=[],
+        help="specification of the channel to import, the format would be "
+        "channel_id=<channel ID>[;<parameter_key>=<parameter_value>;...]. parameter_key "
```

Comment:
This is a cruel command-line format because `;` is the shell statement separator, and `node_ids=blahblah` is a perfectly valid shell command, so I went on a wild goose chase trying to understand why `node_ids` was being ignored!

Reply:
I didn't want to create my own format, the really sad thing is that the alternative to this is to use `&` :)

I suppose I can make it to use `:` as separator instead and a string replace.